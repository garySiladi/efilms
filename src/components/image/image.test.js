// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Image from './image';

const dummyProps = {
  src: 'kobzol',
  alt: 'alt',
  className: 'xxx',
};

test('Image renders with placeholder', () => {
  const component = renderer.create(<Image imageProps={dummyProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Image renders without placeholder', () => {
  const component = renderer.create(<Image imageProps={dummyProps} showPlaceholder={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Image properly handles image loading', () => {
  const imageWrapper = mount(<Image imageProps={dummyProps} />);
  expect(imageWrapper.state().isImageLoaded).toBe(false);
  (imageWrapper.instance(): Object).handleImageLoaded();
  expect(imageWrapper.state().isImageLoaded).toBe(true);
});
