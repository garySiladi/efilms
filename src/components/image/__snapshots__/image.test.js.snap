// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders with placeholder 1`] = `
<div
  className="image image--placeholder"
>
  <div
    className="image__holder"
  >
    <img
      alt="alt"
      className="xxx"
      onLoad={[Function]}
      src="kobzol"
    />
  </div>
</div>
`;

exports[`Image renders without placeholder 1`] = `
<div
  className="image"
>
  <div
    className="image__holder"
  >
    <img
      alt="alt"
      className="xxx"
      onLoad={[Function]}
      src="kobzol"
    />
  </div>
</div>
`;
